### PR TITLE
Ignore no-ops for removing unique constraints

### PR DESCRIPTION
**Pull request by Aergonus**
Addresses #14338

### CODE REVIEW

File: src/south/db/generic.py [hunk 1]
```diff
@@ -639,7 +639,7 @@ def create_unique(self, table_name, columns):
         return name
 
     @invalidate_table_constraints
-    def delete_unique(self, table_name, columns):
+    def delete_unique(self, table_name, columns, if_exist=False):
```
Comment (BYK) [inline src/south/db/generic.py]:
```suggestion
    def delete_unique(self, table_name, columns, required=False):
```

or maybe `optional=true` or `should_exist` etc.?

**Review by BYK — APPROVED** [review]
This won't fix 9.1.2 tho so we may need to release a 9.1.3 for this.

File: src/south/db/generic.py (hunk 1 above)
Comment (Spellchaser) [reply src/south/db/generic.py]: Agreed, should_exist is better name

**Review by BYK — APPROVED** [review]
Still looks good! :)

**Comment by Spellchaser** [comment]
Now that I've been looking at codebase more, I've been wondering why we care if the constraint doesn't exist. The simplest way to handle this would be to change the Database Operations, i.e. https://github.com/getsentry/sentry/blob/master/src/south/db/generic.py#L90 to be `delete_check_sql = 'ALTER TABLE %(table)s DROP CONSTRAINT IF EXISTS %(constraint)s'

**Comment by mattrobenolt** [comment]
How will this apply when we switch off of South? I'm hesitant to change any behaviors in here.

**Comment by Spellchaser** [comment]
I'm not well versed in database/table design but my initial inclination is that if a constraint doesn't exist and we're trying to delete it, it's okay to not fail.

**Comment by mattrobenolt** [comment]
I’m not disagreeing with the behavior. I’m saying, we’re not trying to maintain South for much longer. So if we change the behavior in our vendored South, are we just going to be back to the issue again with a different migrations framework? Or does the newer framework handle this better?

Either way, I think the core of the problem is why don’t you have the index that’s trying to be deleted. It should exist since a previous migration creates it.

**Comment by Spellchaser** [comment]
What happened with my install is that the psql that checked for key constraints depended on the django connection specifying the schema failed. It went to the default schema and found no constraints. Some things worked and other didn't. It ended up being a pretty messy state.

The use case for dying on expected constraints makes debugging clearer

**Pull request closed without merging**